### ISSUE TITLE
Config schema

### ISSUE THREAD

**Issue by michaelwisely** [tymorrow/assigner#3]
We should use [jsonschema](https://pypi.python.org/pypi/jsonschema) to enforce the values in assigner's config file. The config can still be YAML. We just pass a dict to `jsonschema.validate` to validate it.

Maybe something like this....

```python
{                                                                                                                                                                       
    "$schema": "http://json-schema.org/schema#",

    "type": "object",
    "properties": {

        # GitLab domain (https://git.gitlab.com)
        "gitlab-host": {
            "type": "string",
        },

        # GitLab Namespace name
        "namespace": {
            "type": "string",
        },

        # GitLab Namespace ID (we'd have to retrieve that)
        "namespace-id": {
            "type": "integer",
        },

        # Verbose name of the course (might be unnecessary)
        "course-name": {
            "type": "string",
        },

        # Current semester
        "semester": {
            "type": "string",
            "pattern": r"^\d{4}(SP|FS|SM)$"
        },

        # Roster of students
        "roster": {
            "type": "array",
            "items": {
                "type": "object",
                "properties": {

                    # Their full name
                    "name": {
                        "type": "string"
                    },

                    # Section
                    "section": {
                        "type": "string"
                    },

                    # Their GitLab username (single sign on)
                    "username": {
                        "type": "string",
                        "pattern": "^\w+$",
                    },

                    # Their GitLab id (might be handy, but we'd have
                    # to fetch it and save it). Should save time in
                    # the long run instead of constantly querying
                    "gitlab-id": {
                        "type": "integer",
                    },
                },
                "required": ["name", "username", "section"],
                "additionalProperties": False,
            },
        },
    },
    "required": ["gitlab-host", "namespace", "course-name", "semester"],
    "additionalProperties": False,
}
```

**Comment by LinuxMercedes** [comment]
I'm pretty much following this schema as it is.

**Comment by LinuxMercedes** [comment]
Minimal config:
```
gitlab-host: https://git.mst.edu
namespace: 2016-Spring-CS-1001
semester: 2016SP
token: hunter2
```

**Comment by michaelwisely** [comment]
The token! I knew there was an important one that I forgot 😅

**Comment by michaelwisely** [comment]
Closed by #9
